src/modules/flow/http-server/http-server.c

```diff
@@ -190,23 +142,50 @@ boolean_response_cb(void *data, struct sol_http_request *request)
 }
 
 static int
-boolean_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id,
-    const struct sol_flow_packet *packet)
+start_server(struct http_data *http, struct sol_flow_node *node,
+    const char *path)
 {
-    struct http_data *mdata = data;
     int r;
 
-    r = sol_flow_packet_get_boolean(packet, &mdata->value.b);
-    SOL_INT_CHECK(r, < 0, r);
+    if (!server) {
+        server = sol_http_server_new(HTTP_SERVER_PORT);
+        SOL_NULL_CHECK(server, -1);
+    }
 
-    r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
-    SOL_INT_CHECK(r, < 0, r);
+    init_count++;
+    http->path = strdup(path);
+    SOL_NULL_CHECK_GOTO(http->path, err);
+
+    r = sol_http_server_register_handler(server, http->path, common_response_cb, node);
+    SOL_INT_CHECK_GOTO(r, < 0, err_handler);
 
     return 0;
+
+err_handler:
+    free(http->path);
```

Comment:
this start_server is called only from the open(), when it fails open() returns something different from 0 and then the http_data is destroyied, so it's not necessary to set it NULL